src/annotations/components/annotationForm/AnnotationForm.tsx

```diff
@@ -83,64 +103,148 @@ export const AnnotationForm: FC<Props> = (props: Props) => {
   }
 
   const handleKeyboardSubmit = () => {
-    props.onSubmit({message, startTime, endTime})
+    props.onSubmit({
+      summary,
+      startTime,
+      endTime,
+      type: annotationType,
+      id: props.id,
+      stream: props.stream,
+    })
+  }
+
+  // TODO:  get the correct prefix in there, multiple plot types have annotations now
+  const handleDelete = () => {
+    const editedAnnotation = {
+      summary,
+      startTime,
+      endTime,
+      type: annotationType,
+      id: props.id,
+      stream: props.stream,
+    }
+
+    try {
+      dispatch(deleteAnnotations(editedAnnotation))
+      event('annotations.delete_annotation.success')
+      dispatch(notify(deleteAnnotationSuccess(editedAnnotation.summary)))
+      props.onClose()
+    } catch (err) {
+      event('annotations.delete_annotation.failure')
+      dispatch(notify(deleteAnnotationFailed(err)))
+    }
   }
 
   const handleCancel = () => {
     event('dashboards.annotations.create_annotation.cancel')
     props.onClose()
   }
 
+  const changeToRangeType = () => {
+    setAnnotationType('range')
+    if (!endTime) {
+      setEndTime(startTime)
+    }
+  }
+
+  const changeToPointType = () => {
+    setAnnotationType('point')
+  }
+
+  const saveTextSuffix = props.id ? 'Changes' : 'Annotation'
+
+  const footerClasses = classnames('annotation-form-footer', {
+    'edit-annotation-form-footer': props.id,
+  })
+
+  const buttonClasses = classnames({'edit-annotation-buttons': props.id})
+
   return (
     <Overlay.Container maxWidth={ANNOTATION_FORM_WIDTH}>
       <Overlay.Header
         title={`${props.title} Annotation`}
         onDismiss={handleCancel}
+        className="edit-annotation-head"
       />
       <Form onSubmit={handleSubmit}>
         <Overlay.Body>
-          <Grid>
-            <Grid.Row>
+          <Form.Label label="Type" />
+          <SelectGroup
+            size={ComponentSize.Small}
+            style={{marginBottom: 8}}
+            color={ComponentColor.Default}
+          >
+            <SelectGroup.Option
+              onClick={changeToPointType}
+              active={'point' === annotationType}
+              id="annotation-form-point-type"
+              value="point"
+            >
+              Point
+            </SelectGroup.Option>
+            <SelectGroup.Option
+              onClick={changeToRangeType}
+              value="range"
+              active={'range' === annotationType}
+              id="annotation-form-range-type"
+            >
+              Range
+            </SelectGroup.Option>
+          </SelectGroup>
+          <div style={{display: 'flex', width: '100%'}}>
```

Comment:
I think this, with the way it is, achieves the visuals we want in the UI.

 The way I'm reading this CSS is :
- When there is only 1 annotation time input, it stretches across 100% of the available width,
- When there are two annotation time Input, The second input appears to the right of the first one and these two combined stretches across 100% of the available width.

I think you could also use display: grid, grid-template-column: 50% 50%; column-gap: 10px; to get the same effect.

Reply:
Would it be cleaner to make this a class rather than inline css? 
Just a thought. If you have a different convention for the codebase, feel free to ignore me

Reply:
i agree a class is easier than inline css, changing it to that.